sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/SqlQuerySpec.java

```diff
@@ -70,7 +71,7 @@ public SqlQuerySpec(String queryText, List<SqlParameter> parameters) {
     public SqlQuerySpec(String queryText, SqlParameter... parameters) {
         this.jsonSerializable = new JsonSerializable();
         this.setQueryText(queryText);
-        this.parameters = Arrays.asList(parameters);
+        this.parameters = Collections.synchronizedList(Arrays.asList(parameters));
```

Comment:
why this needs to be synchronized. If Encryption is using this it has to make the needed changes on top of this to ensure thread-safety.

why does this in the core sdk need to be synchrnozied? why can't the synchronization move to the encryption layer?

Reply:
Please see the reply for addSqlParameter Api "Currently due to the lack of this api , it was very inefficient on encryption , to consume and recreate the array on every add encryption parameter. I don't see any reason for not having this api, please let me know if there is any which i am missing . Also it is a helpful api for regular cosmos user. Added beta annotation"
Once we have this addSqlParameter Api, it is good to have a synchronized list. For encryption I can fire the multiple encryption parameter code, which is more efficient instead of chaining one by one. We could also have the similar scenario for non encryption users